### ISSUE TITLE
Promotion menu empty

### ISSUE THREAD

**Issue by kraktus** [veloce/lichobile#1793]
Reported here: https://lichess.org/forum/lichess-feedback/bug-unable-to-promote-pawn#1
> The field there i should be able to choose figure is empty. First time i found this bug a few months ago and it repeats every time im trying to promote my pawns. That's why i hate late game 😅
I've already tried to reinstall lichess app, but that didn't solve the problem.
Also, I've got no problems in puzzles — I'm able to promote pawn there, so i have no idea why this field is empty in normal games

![](https://i.ibb.co/3vz9nFK/Screenshot-2021-08-06-19-36-52-219-org-lichess-mobileapp.jpg)
Android 8.1.0
redmi 6
lichess app version 7.11.3 (latest)

Maybe fixed by https://github.com/veloce/lichobile/pull/1785 ?

**Comment by bja555** [comment]
Same problem with my lichess account bja555

**Comment by jas14** [comment]
From #2170:
> Device: nexus 7 2013
> OS: android 7.1.2

Also reported in #2179 . @matejplank, as requested in the issue template, please provide:

- Device make and model
- OS version
- Lichess version

**Comment by indiebubbler** [comment]
I want to report this bug occurs still on:
Device: LG G6
OS: Android 9
Lichess version: 7.16.0

**Comment by indiebubbler** [comment]
guys, this is kind of showstopper, isn't it?

**Comment by jas14** [comment]
Sorry, we can't fix it if we can't reproduce it.

**Comment by hackerONrails** [comment]
@jas14 I had it on Xiami Mi A2, Android 10. 

I installed now 7.15.2 to be able to promote again:
https://github.com/lichess-org/lichobile/releases/tag/v7.15.2

**Comment by williamAlhant** [comment]
This bug just happened to me. I'm on v7.16.0

**Comment by eljamm** [comment]
As described in #2191, updating the Android WebView to the latest version fixed the issue for my device.
Those who can't update can test [this version](https://github.com/eljamm/lichobile/releases/download/latest/app-debug.apk), which uses fallback values for the promo menu.

Hope this solves the issue for everyone.

**Comment by williamAlhant** [comment]
Is not Chrome used instead of Webview, or is it the same? I cannot update Webview on my phone but I can update Chrome, of which I probably had an outdated version since it's not my browser of choice.

**Comment by eljamm** [comment]
Well, according to the [google docs](https://developer.chrome.com/docs/multidevice/webview/#does-this-mean-chrome-for-android-is-using-the-webview), chrome and webview aren't exactly the same thing on Android. Also, the README mentioned being `close to the android webview`, so I think that's what's being used here.

For my device, I was on WebView 70 and updating it to WebView 105 fixed the issue for me. Can you tell me which version of WebView your device is using ? It should be named `Android System WebView` in your App List.

If Google's Play Store doesn't show the update you can try using an alternative such as the [Aurora Store](https://f-droid.org/en/packages/com.aurora.store/). If you still can't update, you could try [this release](https://github.com/eljamm/lichobile/releases/download/latest/app-debug.apk), which tries to fix this issue for old versions of WebView.

**Comment by williamAlhant** [comment]
Actually, updating Chrome seems to solve the issue for me. My Chrome version was in the 70s, probably 72 (that's the version I have when I uninstall the updates in the app settings). I don't have the bug when it is updated to the latest version (105).

**Comment by eljamm** [comment]
I only have WebView installed on my device so I didn't try with chrome, but if updating it also fixes the issue than that's great to hear.

**Comment by vadz** [comment]
I've got it on Mi 9T Pro with Android 10 with v7.16. Downgrading to v7.15.2 fixed the problem, so something definitely changed in the latest version.

**Comment by lgrapenthin** [comment]
same problem on android 8.1.0 and latest lichess app.

**Comment by jas14** [comment]
#2191 will probably fix this issue. The other option is to revert #2111, which will probably be simpler but will reopen #2074.

**Comment by ornicar** [comment]
Getting more emails about this bug every day.

**Comment by veloce** [comment]
Right. So I'll release the fix this week, hoping it'll solve the issue.

**Comment by AlexisTM** [comment]
I got the same issue recently on a Galaxy Tab S3, the app's last version. 

Unlike in the actual game where it happened, in over-the-board mode I couldn't exit that state without pressing the "return" button. 

![Screenshot_20230518-232404_lichess](https://github.com/lichess-org/lichobile/assets/6976744/a8556f1a-ceb7-4df3-a872-e9f9e7d45da5)

**Comment by edquist** [comment]
Hi, after updating to v8.0.0, on a Moto G Power running android 10, I got the blank promotion menu when promoting *in analysis mode*.  (But promoting during normal play works.)  Had to downgrade the app to v7.15.2 to get that to work again.

Did promotion during analysis mode get missed in the workaround?

Thanks!

**Comment by endriu11111111** [comment]
"I also have a problem with promoting to queen in the Lichess game in the Android app. The application version is 8.0.0, Android version 10, phone model: Motorola One (XT-1941-4).

Please help me kindly."

![Screenshot_20230726-143507.png](https://github.com/lichess-org/lichobile/assets/140278579/65d023dd-dddc-4f78-80d2-037d3e6bedf8)

**Comment by plaf-git** [comment]
Hi, updating Android System Webview from v76 to v95 fixed this problem on my quite old phone (LG G2 mini, Android 5.0.2), thanks for the solution !
